composeApp/src/commonMain/kotlin/org/ooni/probe/di/Dependencies.kt

```diff
@@ -312,6 +331,30 @@ class Dependencies(
             uploadMissingMeasurements = uploadMissingMeasurements::invoke,
         )
 
+    fun addDescriptorViewModel(
+        descriptorId: String,
+        onBack: () -> Unit,
+    ) = AddDescriptorViewModel(
+        onCancel = {
+            // TODO(aanorbel): show toast/snackbar
+            onBack()
+        },
+        onError = {
+            // TODO(aanorbel): show toast/snackbar
+            onBack()
+        },
+        saveTestDescriptors = {
+            saveTestDescriptors.invoke(it)
+            // TODO(aanorbel): show toast/snackbar
+            onBack()
+        },
+        fetchDescriptor = {
+            withContext(backgroundDispatcher) {
```

Comment:
You should hide this detail inside the `fetchDescriptor`, but the `engine.httpDo` call is already running on the backgroundDispatcher.